src/tree.rs

```diff
@@ -660,6 +661,30 @@ impl<T> Node<T> {
     }
 }
 
+impl<T> Node<T> {
+    /// Iterates over the tree and calls the given visitor function
+    /// with fully resolved path and its value.
+    pub fn for_each<V: FnMut(String, T) -> bool>(&self, mut visitor: V) {
+        let mut queue = VecDeque::new();
+        queue.push_back((self.prefix.clone(), self));
+        while let Some((mut prefix, node)) = queue.pop_front() {
+            denormalize_params(&mut prefix, &node.remapping);
+            if node.value.is_some() {
```

Comment:
Small nit, this could be simplified as `if let Some(value) = &node.value` to avoid the unwrap.